### PR TITLE
Add note in Float.to_{string/charlist}/1

### PR DESCRIPTION
**Pull request by v0idpwn**
Makes it clear where to look for if you're looking to use something
other than the algorithm used there.

### CODE REVIEW

**Comment by v0idpwn** [comment]
Also, I believe the statement above will be imprecise starting from OTP 25. Shall we fix it already?

**Comment by josevalim** [comment]
The statement is false from OTP 24, so amending it will be welcome. Here is the paper: https://dl.acm.org/citation.cfm?doid=3296979.3192369

**Comment by v0idpwn** [comment]
Though it was merged to master a while ago, it doesn't seem like it was released.

**Comment by josevalim** [comment]
OTP 24 has the algorithm implemented in Erlang, which is used by io_lib. 25 brings the algorithm to C.

File: lib/elixir/lib/float.ex [hunk 1]
```diff
@@ -509,13 +509,20 @@ defmodule Float do
   defp sign(1, num), do: -num
 
   @doc """
-  Returns a charlist which corresponds to the text representation
+  Returns a charlist which corresponds to the smallest text representation
```
Comment (josevalim) [inline lib/elixir/lib/float.ex:512]:
```suggestion
  Returns a charlist which corresponds to the shortest text representation
```

File: lib/elixir/lib/float.ex [hunk 2]
```diff
@@ -529,13 +536,20 @@ defmodule Float do
   end
 
   @doc """
-  Returns a binary which corresponds to the text representation
+  Returns a binary which corresponds to the smallest text representation
```
Comment (josevalim) [inline lib/elixir/lib/float.ex:539]:
```suggestion
  Returns a binary which corresponds to the shortest text representation
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**